### DISCUSSION TITLE
Django User not created

### DISCUSSION THREAD

**Discussion by noman-ejaz** [community/community#52197] [Programming Help]
### Select Topic Area

Bug

### Body

I ma working on an inventory management system and i have downloaded a template for front end "sneat". After this i have setup it according to django file formate. I need to setup its signup and login for user(who manages the store). I have write code for this but user is not creating and not showing in database(dbsqlite3) this is happening first time to me,i know it is easy but i don't know why it is happening.
My signup.html file is..
`<form id="formAuthentication" class="mb-3" action="Login" method="POST">
                {% csrf_token %}
                
                <div class="mb-3">
                  <label for="username" class="form-label">Username</label>
                  <input
                    type="text"
                    class="form-control"
                    id="username"
                    name="username"
                    placeholder="Enter your username"
                    autofocus
                    required
                  />
                </div>
                <div class="mb-3">
                  <label for="email" class="form-label">Email</label>
                  <input type="text" class="form-control" id="email" name="email" placeholder="Enter your email" required />
                </div>
                <div class="mb-3 form-password-toggle">
                  <label class="form-label" for="password">Password</label>
                  <div class="input-group input-group-merge">
                    <input
                      type="password"
                      id="password"
                      class="form-control"
                      name="password"
                      placeholder="&#xb7;&#xb7;&#xb7;&#xb7;&#xb7;&#xb7;&#xb7;&#xb7;&#xb7;&#xb7;&#xb7;&#xb7;"
                      aria-describedby="password"
                      required
                    />
                    <span class="input-group-text cursor-pointer"><i class="bx bx-hide"></i></span>
                  </div>
                </div>

                <div class="mb-3">
                  <div class="form-check">
                    <input class="form-check-input" type="checkbox" id="terms-conditions" name="terms" />
                    <label class="form-check-label" for="terms-conditions">
                      I agree to
                      <a href="javascript:void(0);">privacy policy & terms</a>
                    </label>
                  </div>
                </div>
                <button class="btn btn-primary d-grid w-100" type="submit">Sign up</button>
              </form>

              <p class="text-center">
                <span>Already have an account?</span>
                <a href="Login">
                  <span>Sign in instead</span>
                </a>
              </p>
            </div>
          </div>
          
        </div>
      </div>
    </div>`

My views are...

`def Signup(request):
    if request.method == 'POST':
        username = request.POST.get('username')
        email = request.POST.get('email')
        password = request.POST.get('password')
        
        # Check all required fields are filled
        if not username or not email or not password:
            messages.error(request, 'Please fill all required fields.')
            return redirect('Signup')

        # Check email is not already in use
        if User.objects.filter(email=email).exists():
            messages.error(request, 'Email already exists.')
            return redirect('Signup')

        # Check email syntax is correct
        try:
            validate_email(email)
        except ValidationError:
            messages.error(request, 'Please enter a valid email address.')
            return redirect('Signup')

        # Create new user account
        user = User.objects.create_user(username=username, email=email, password=password)
        user.save()

        # Redirect to login page
        messages.success(request, 'Your account has been created. Please log in.')
        return redirect('Login')

    return render(request, 'html/auth-register-basic.html')`

and my urls are...

`from django.urls import path
from login import views

urlpatterns=[
    path("",views.Index,name="Index"),
    path("Login",views.Login,name="Login"),
    path("Register",views.Register,name="Register"),
    path("forgetPassword",views.forgetPassword,name='forgetPassword'),
    path("Signup",views.Signup,name='Signup'),
    path("loginForm",views.loginForm,name="loginForm"),
    path("Account",views.Account,name="Account")
    
    
]`

**Comment by xqzmes** [comment]
Hey!
I advise you to create a separate file in which to make a form. It is a very useful tool. I think it will help you in this case.
here is - https://docs.djangoproject.com/en/4.2/topics/forms/

**Comment by cehfisher** [comment]
Hi @noman-ejaz - It looks like your issue was mistakenly added to the accessibility discussion board. @queenofcorgis could you please move this to the correct category? Thanks!
